jenkins/test-history/gen_json.py

```diff
@@ -23,198 +23,271 @@
-        yield subpath
-
-def gcs_ls_build(job, build):
-    """Lists all files under a given job and build path."""
-    url = 'gs://kubernetes-jenkins/logs/{}/{}'.format(job, str(build))
-    for path in gcs_ls(url):
-        yield path
-
-
-def gcs_ls_artifacts(job, build):
-    """Lists all artifacts for a build."""
-    for path in gcs_ls_build(job, build):
-        if path.endswith('artifacts/'):
-            for artifact in gcs_ls(path):
-                yield artifact
-
-
-def gcs_ls_junit_paths(job, build):
-    """Lists the paths of JUnit XML files for a build."""
-    for path in gcs_ls_artifacts(job, build):
-        if re.match(r'.*/junit.*\.xml$', path):
-            yield path
-
-
-def gcs_get_tests(path):
-    """Generates test data out of the provided JUnit path.
-
-    Returns None if there's an issue parsing the XML.
-    Yields name, time, failed, skipped for each test.
-    """
-    try:
-        data = subprocess.check_output(
-            ['gsutil', 'cat', path], stderr=open(os.devnull, 'w'))
-    except subprocess.CalledProcessError:
-        return
-
-    try:
-        data = zlib.decompress(data, zlib.MAX_WBITS | 16)
-    except zlib.error:
-        # Don't fail if it's not gzipped.
-        pass
-
-    try:
-        root = ElementTree.fromstring(data)
-    except ElementTree.ParseError:
-        return
-
-    for child in root:
-        name = child.attrib['name']
-        ctime = float(child.attrib['time'])
-        failed = False
-        skipped = False
-        for param in child:
-            if param.tag == 'skipped':
-                skipped = True
-            elif param.tag == 'failure':
-                failed = True
-        yield name, ctime, failed, skipped
-
-
-def get_tests_from_junit_path(path):
-    """Generates all tests in a JUnit GCS path."""
-    for test in gcs_get_tests(path):
-        if not test:
-            continue
-        yield test
-
-
-def get_tests_from_build(job, build):
-    """Generates all tests for a build."""
-    for junit_path in gcs_ls_junit_paths(job, build):
-        for test in get_tests_from_junit_path(junit_path):
-            yield test
-
-
-def get_daily_builds(server, matcher):
-    """Generates all (job, build) pairs for the last day."""
-    now = time.time()
-    for job in get_jobs(server):
-        if not matcher(job):
-            continue
-        for build in reversed(sorted(get_builds(server, job))):
-            building, timestamp = get_build_info(server, job, build)
-            # Skip if it's still building.
-            if building:
-                continue
-            # Quit once we've walked back over a day.
-            if now - timestamp / 1000 > 60*60*24:
-                break
-            yield job, build
 
+import multiprocessing
+import requests
+
+
+class GCSClient(object):
+
+    def __init__(self, jobs_dir):
+        self.jobs_dir = jobs_dir
+        self.session = requests.Session()
+
+    def request(self, path, params, json=True):
```

Comment:
I'm wary of an argument with the same name as a module. Is that okay? @fejta should we turn on pylint for this repo?